src/devices/machine/spg110_video.cpp

```diff
@@ -475,33 +506,44 @@ void spg110_video_device::map_video(address_map &map)
 	map(0x04200, 0x043ff).ram().share("sprattr1");
 	map(0x04400, 0x045ff).ram().share("sprattr2");
 
-	map(0x08000, 0x081ff).ram().share("palram"); // palette format unknown
+	map(0x08000, 0x081ff).ram().w(FUNC(spg110_video_device::palette_w)).share("palram"); // palette format unknown
 }
 
 void spg110_video_device::device_start()
 {
 	save_item(NAME(m_dma_src_step));
 	save_item(NAME(m_dma_dst_step));
 	save_item(NAME(m_dma_unk_2061));
-	save_item(NAME(m_dma_unk_2067));
+	save_item(NAME(m_dma_src_high));
 	save_item(NAME(m_dma_dst));
 	save_item(NAME(m_dma_src));
 	save_item(NAME(m_bg_scrollx));
 	save_item(NAME(m_bg_scrolly));
 	save_item(NAME(m_2036_scroll));
+
+	m_video_irq_cb.resolve();
+
+	if (!strcmp(machine().system().name, "jak_spdmo"))
```

Comment:
Please don't do this. strcmp hacks are just awful.

Reply:
yeah I know, it's not intended to stay tho, and is clearly something that should be configured by the registers, therefore passing it down from another level (eg. driver) makes no sense as that would just be more code to remove later.

in the context of what is being done, this actually seemed like the cleanest / simplest solution for a temporary workaround in order to further improve the video emulation.